heartbeat/processor/crossShardStatusProcessor.go

```diff
@@ -0,0 +1,166 @@
+package processor
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go/heartbeat"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+// ArgCrossShardStatusProcessor represents the arguments for the cross shard status processor
+type ArgCrossShardStatusProcessor struct {
+	Messenger            p2p.Messenger
+	PeerShardMapper      process.PeerShardMapper
+	ShardCoordinator     sharding.Coordinator
+	DelayBetweenRequests time.Duration
+}
+
+type crossShardStatusProcessor struct {
+	messenger            p2p.Messenger
+	peerShardMapper      process.PeerShardMapper
+	shardCoordinator     sharding.Coordinator
+	delayBetweenRequests time.Duration
+	cancel               func()
+	// todo remove this - tests only
+	LatestKnownPeers map[string][]core.PeerID
+}
+
+// NewCrossShardStatusProcessor creates a new instance of crossShardStatusProcessor
+func NewCrossShardStatusProcessor(args ArgCrossShardStatusProcessor) (*crossShardStatusProcessor, error) {
```

Comment:
we plan to phase out this new component. We will try to do in a next PR a mechanism for the peers to share the shard ID at the connect time.